velox/exec/StreamingWindow.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/StreamingWindow.h"
+#include "velox/exec/OperatorUtils.h"
+#include "velox/exec/Task.h"
+
+namespace facebook::velox::exec {
+
+void StreamingWindow::updatePartitions() {
+  partitionStartRows_.push_back(sortedRows_.size());
+  sortedRows_.insert(
+      sortedRows_.end(), partitionRows_.begin(), partitionRows_.end());
+  partitionRows_.clear();
+}
+
+void StreamingWindow::addInput(RowVectorPtr input) {
+  for (auto col = 0; col < input->childrenSize(); ++col) {
+    decodedInputVectors_[col].decode(*input->childAt(col));
+  }
+
+  auto partitionCompare = [&](const char* lhs, const char* rhs) -> bool {
```

Comment:
This variable is not needed. Call compareRowsWithKeys directly before for readability.

Reply:
Updated.